backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/EntryPoint.kt

```diff
@@ -0,0 +1,93 @@
+import org.jetbrains.kotlin.backend.common.descriptors.WrappedValueParameterDescriptor
+import org.jetbrains.kotlin.backend.common.lower.createIrBuilder
+import org.jetbrains.kotlin.backend.common.lower.irBlockBody
+import org.jetbrains.kotlin.backend.konan.ir.buildSimpleAnnotation
+import org.jetbrains.kotlin.descriptors.Modality
+import org.jetbrains.kotlin.descriptors.Visibilities
+import org.jetbrains.kotlin.ir.UNDEFINED_OFFSET
+import org.jetbrains.kotlin.ir.builders.*
+import org.jetbrains.kotlin.ir.declarations.IrDeclarationOrigin
+import org.jetbrains.kotlin.ir.declarations.IrFunction
+import org.jetbrains.kotlin.ir.declarations.impl.IrFunctionImpl
+import org.jetbrains.kotlin.ir.declarations.impl.IrValueParameterImpl
+import org.jetbrains.kotlin.ir.expressions.impl.IrTryImpl
+import org.jetbrains.kotlin.ir.symbols.impl.IrSimpleFunctionSymbolImpl
+import org.jetbrains.kotlin.ir.symbols.impl.IrValueParameterSymbolImpl
+import org.jetbrains.kotlin.ir.types.typeWith
+import org.jetbrains.kotlin.ir.util.irCatch
+import org.jetbrains.kotlin.name.Name
+
+internal fun makeEntryPoint(context: Context): IrFunction {
+    val entryPointDescriptor = WrappedSimpleFunctionDescriptor()
+    val actualMain = context.ir.symbols.entryPoint!!.owner
+    val entryPoint = IrFunctionImpl(
+            actualMain.startOffset,
+            actualMain.startOffset,
+            IrDeclarationOrigin.DEFINED,
+            IrSimpleFunctionSymbolImpl(entryPointDescriptor),
+            Name.identifier("Konan_start"),
+            Visibilities.PRIVATE,
+            Modality.FINAL,
+            context.irBuiltIns.intType,
+            isInline = false,
+            isExternal = false,
+            isTailrec = false,
+            isSuspend = false
+    ).also { function ->
+        function.valueParameters.add(WrappedValueParameterDescriptor().let {
+            IrValueParameterImpl(
+                    actualMain.startOffset, actualMain.startOffset,
+                    IrDeclarationOrigin.DEFINED,
+                    IrValueParameterSymbolImpl(it),
+                    Name.identifier("args"),
+                    index = 0,
+                    varargElementType = null,
+                    isCrossinline = false,
+                    type = context.irBuiltIns.arrayClass.typeWith(context.irBuiltIns.stringType),
+                    isNoinline = false
+            ).apply {
+                it.bind(this)
+                parent = function
+            }
+        })
+    }
+    entryPointDescriptor.bind(entryPoint)
+    entryPoint.annotations += buildSimpleAnnotation(context.irBuiltIns,
+            actualMain.startOffset, actualMain.startOffset,
+            context.ir.symbols.exportForCppRuntime.owner, "Konan_start")
+
+    val builder = context.createIrBuilder(entryPoint.symbol)
+    entryPoint.body = builder.irBlockBody(entryPoint) {
+        +IrTryImpl(
+                startOffset = actualMain.startOffset,
+                endOffset   = actualMain.startOffset,
```

Comment:
Is this intended?

Reply:
Yes, here and below we just set entry point as if it was in very beginning of the main.